File: python/director/temperature-check/handler_test.py
import json
from types import SimpleNamespace

try:
    from . import handler as h
except ImportError:
    import handler as h


def test_sets_alert_above_threshold():
    event = SimpleNamespace(body=json.dumps({"temperature_c": 82.4}).encode())
    response = h.handle(event, {})

    assert response["statusCode"] == 200
    assert response["body"]["alert"] is True
    assert response["body"]["threshold_c"] == 75.0

File: node/director/temperature-check/handler.test.js
'use strict'

const assert = require('node:assert/strict')
const { test } = require('node:test')

const handler = require('./handler')

test('raises an alert above the temperature threshold', async () => {
  const context = new TestContext()
  await handler({ body: { temperature_c: 82.4 } }, context)

  assert.equal(context.statusCode, 200)
  assert.deepEqual(context.body, {
    value_c: 82.4,
    threshold_c: 75,
    alert: true
  })
})

class TestContext {
  status (value) {
    this.statusCode = value
    return this
  }

  headers (value) {
    this.headerValues = value
    return this
  }

  succeed (value) {
    this.body = value
    return value
  }
}
